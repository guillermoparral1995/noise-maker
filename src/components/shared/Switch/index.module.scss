.label {
    bottom: 8px;
    font-size: 12px;
    color: rgba(255, 255, 255, 0.6);
    margin-right: 5px;
    position: relative;
}