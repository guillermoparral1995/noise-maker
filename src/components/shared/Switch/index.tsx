import { switchValues } from '@constants/switchValues';
import { ActionTypes, Switchs } from '@types';
import { InputSwitch, InputSwitchChangeEvent } from 'primereact/inputswitch';
import React from 'react';
import styles from './index.module.scss';

const Switch = ({
  id,
  value,
  dispatch,
}: {
  id: Switchs;
  value: boolean;
  dispatch: React.Dispatch<ActionTypes>;
}) => {
  const handleToggle = (e: InputSwitchChangeEvent) => {
    dispatch(switchValues[id].action(e.value));
  };

  return (
    <div data-testid={id}>
      <label className={styles.label} htmlFor={id}>
        {switchValues[id].label}
      </label>
      <InputSwitch
        pt={{
          hiddenInput: {
            id,
          },
        }}
        checked={value}
        onChange={handleToggle}
      />
    </div>
  );
};

export default Switch;
