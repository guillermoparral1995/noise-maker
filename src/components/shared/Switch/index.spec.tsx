import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { Actions, Switchs } from '@types';
import Switch from '.';

describe('Switch', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render with switch label', async () => {
    const { findByRole } = render(
      <Switch id={Switchs.DELAY} value={false} dispatch={jest.fn()}></Switch>,
    );

    const delaySwitch = await findByRole('checkbox');
    expect(screen.getByText('Delay')).toBeVisible();
    expect(delaySwitch).toMatchSnapshot();
  });

  it('should call dispatch when clicking', async () => {
    const mockDispatch = jest.fn();
    const { findByRole } = render(
      <Switch
        id={Switchs.DELAY}
        value={false}
        dispatch={mockDispatch}
      ></Switch>,
    );

    const delaySwitch = await findByRole('checkbox');
    fireEvent.click(delaySwitch);

    expect(mockDispatch).toHaveBeenCalledWith({
      payload: true,
      type: Actions.SWITCH_DELAY,
    });
  });
});
