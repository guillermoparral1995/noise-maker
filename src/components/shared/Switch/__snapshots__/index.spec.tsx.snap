// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render with switch label 1`] = `
<div
  aria-checked="false"
  class="p-inputswitch p-component"
  data-pc-name="inputswitch"
  data-pc-section="root"
  role="checkbox"
>
  <div
    class="p-hidden-accessible"
    data-pc-section="hiddeninputwrapper"
  >
    <input
      aria-checked="false"
      data-pc-section="hiddeninput"
      id="DELAY"
      role="switch"
      type="checkbox"
    />
  </div>
  <span
    class="p-inputswitch-slider"
    data-pc-section="slider"
  />
</div>
`;
